graylog2-server/src/main/java/org/graylog2/indexer/migration/RemoteReindexMigration.java

```diff
@@ -92,15 +94,23 @@ public int progress() {
             return 100; // avoid division by zero. No indices == migration is immediately done
         }
 
-        final double indexPortion = 100.0 / indices.size();
-
-        final double overallProgress = indices.stream()
+        final BigDecimal sum = indices.stream()
                 .filter(i -> i.progress() != null)
-                .mapToDouble(i -> i.progress().progressPercent() / 100.0)
-                .map(relativeProgress -> relativeProgress * indexPortion)
-                .sum();
+                .map(RemoteReindexMigration::indexProgress)
+                .reduce(BigDecimal.ZERO, BigDecimal::add);
+
+        return sum.divide(BigDecimal.valueOf(indices.size()), 4, RoundingMode.HALF_UP).scaleByPowerOfTen(2).intValue();
```

Comment:
could this lead to a wrong result since `i.progress() == null` is filtered out above and included here by simply using `size()`?

Reply:
no, I see. if they don't have progress they shouldn't be accounted for in the percentage anyway and be treated like 0 progress.

Reply:
yes, exactly